[web] Rename, organize and implement lazy loading in home sections

derianrddev opened the pull request:
### **Overview**

This PR introduces significant updates to the home page by renaming sections for better clarity, reorganizing components according to the **Code Style and Conventions Guide**, and implementing lazy loading for non-critical sections to enhance performance.

### **Key Changes**

1. **Renamed Components:**
    - Updated section names to better reflect their purpose.
2. **Component Organization:**
    - All components were organized following the **Code Style and Conventions Guide**:
        - Location: `~/components/sections/home/`.
        - Modular structure for better scalability, clarity, and maintainability.
3. **Lazy Loading Implementation:**
    - Used `next/dynamic` to load the home sections lazily.
    - Ensured immediate loading of critical sections like `Hero` to maintain a smooth user experience.

### **Issue Reference**

Closes #38 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

# Release Notes

- **New Features**
    - Completely redesigned home page with dynamic component loading.
    - Added new sections: Hero, User Journey, Highlighted Projects, Join Us, How It Works, New User Guide, Platform Overview, Community, and Final CTA.
    - Introduced dynamic animations and interactive UI elements.

- **Bug Fixes**
    - Corrected the naming of the `Contribute` component for consistency.

- **Refactoring**
    - Removed multiple static section components, including CommunitySection, CreatorSection, and others.
    - Replaced static imports with dynamic Next.js imports.
    - Enhanced component structure and modularity.

- **Performance**
    - Implemented lazy loading for home page sections.
    - Added loading states for dynamically imported components.

- **Design Updates**
    - Improved visual consistency across home page sections.
    - Added motion and animation effects.
    - Enhanced responsive design.

- **Mock Data**
    - Created comprehensive mock data for various sections.
    - Introduced type-safe interfaces for data structures.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

AndlerRL commented on apps/web/components/sections/home/hero.tsx:
```diff
@@ -0,0 +1,276 @@
+    label: 'Healthcare Support',
+    color: 'border-cyan-200/50 text-cyan-700 hover:bg-cyan-50/80',
+  },
+  {
+    id: 'food-security-id',
+    icon: <UtensilsCrossed className="w-4 h-4" />,
+    label: 'Food Security Campaigns',
+    color: 'border-orange-200/50 text-orange-700 hover:bg-orange-50/80',
+  },
+  {
+    id: 'child-welfare-id',
+    icon: <Baby className="w-4 h-4" />,
+    label: 'Child Welfare Programs',
+    color: 'border-purple-200/50 text-purple-700 hover:bg-purple-50/80',
+  },
+  {
+    id: 'sustainable-agriculture-id',
+    icon: <Sprout className="w-4 h-4" />,
+    label: 'Sustainable Agriculture',
+    color: 'border-emerald-200/50 text-emerald-700 hover:bg-emerald-50/80',
+  },
+  {
+    id: 'social-finance-id',
+    icon: <Coins className="w-4 h-4" />,
+    label: 'Social Finance & Innovation',
+    color: 'bg-sky-50/80 text-sky-700 hover:bg-sky-100/80 border-sky-200/50',
+  },
+  {
+    id: 'education-for-all-id',
+    icon: <GraduationCap className="w-4 h-4" />,
+    label: 'Education for All',
+    color: 'bg-indigo-50/80 text-indigo-700 hover:bg-indigo-100/80 border-indigo-200/50',
+  },
+  {
+    id: 'disaster-relief-id',
+    icon: <HandHelping className="w-4 h-4" />,
+    label: 'Disaster Relief Efforts',
+    color: 'bg-red-50/80 text-red-700 hover:bg-red-100/80 border-red-200/50',
+  },
+];
```
These can be at the `lib/mock-data/mock-categories.ts` file.

@coderabbitai always suggest to move to that folder and follow that file pattern that I suggested above for every mocked data in the webapp. It is ok for now to have mock data.

AndlerRL commented on apps/web/components/sections/home/final-cta.tsx:
```diff
@@ -0,0 +1,216 @@
+import { SocialButton } from '~/components/shared/social-cta';
+
+// Constants
+const features: Feature[] = [
+  {
+    id: 'feature-1',
+    icon: <Zap className="w-6 h-6 text-blue-700" />,
+    title: 'Intuitive Web3 Interface',
+    description:
+      'We simplify blockchain interaction for both creators and contributors, offering a seamless, user-friendly experience with digital wallets and smart contracts.',
+  },
+  {
+    id: 'feature-2',
+    icon: <Target className="w-6 h-6 text-purple-700" />,
+    title: 'Diversified Impact',
+    description:
+      'Unlock multiple ways to collaborate using Web3 wallets, maximizing social impact and expanding opportunities for meaningful change.',
+  },
+  {
+    id: 'feature-3',
+    icon: <Shield className="w-6 h-6 text-blue-700" />,
+    title: 'Blockchain Transparency',
+    description:
+      'Every project is traceable in real-time, showcasing direct impact on the blockchain. This ensures all participants trust that their support reaches its intended destination.',
+  },
+  {
+    id: 'feature-4',
+    icon: <Settings className="w-6 h-6 text-purple-700" />,
+    title: 'Decentralized Verification with Smart Escrow',
+    description:
+      'Through a secure escrow system and decentralized validation, we guarantee that every project’s funds are protected and released only when conditions are met.',
+  },
+];
```
Same observation as this comment -> https://github.com/kindfi-org/kindfi/pull/82#discussion_r1929893420

AndlerRL commented on apps/web/components/pages/home.tsx:
```diff
@@ -10,167 +1,27 @@
-			<LatamWeb3Platform />
-			<CommunitySection />
-			<CreatorSection />
+			<UserJourney />
+			<HighlightedProjects />
+			<JoinUs />
+			<HowItWorks />
+			<NewUserGuide />
+			<PlatformOverview />
+			<Community />
+			<FinalCTA />
```
@coderabbitai the layout component would have those tags. for inner page components we only worry about the inner sections.

AndlerRL commented on apps/web/components/sections/home/highlighted-projects.tsx:
```diff
@@ -0,0 +1,231 @@
+      { id: "community-tag-id", text: "COMMUNITY" },
+    ],
+  },
+  {
+    id: "empowering-education-id",
+    image: "/images/education.webp",
+    category: "Education",
+    title: "Empowering Education",
+    description:
+      "Support education programs for children in low-income areas. Together, we can bridge the education gap and create opportunities for the next generation.",
+    currentAmount: 40000,
+    targetAmount: 55000,
+    investors: 40,
+    minInvestment: 10,
+    percentageComplete: 73,
+    tags: [
+      { id: "education-tag-id", text: "EDUCATION" },
+      { id: "children-tag-id", text: "CHILDREN" },
+      { id: "future-tag-id", text: "FUTURE" },
+    ],
+  },
+  {
+    id: "mobile-clinics-id",
+    image: "/images/healthcare.webp",
+    category: "Healthcare",
+    title: "Mobile Clinics",
+    description:
+      "Bring essential healthcare services to remote areas through mobile clinics. Your support helps save lives and build healthier communities.",
+    currentAmount: 32000,
+    targetAmount: 45000,
+    investors: 30,
+    minInvestment: 20,
+    percentageComplete: 71,
+    tags: [
+      { id: "healthcare-tag-id", text: "HEALTHCARE" },
+      { id: "community-tag-id", text: "COMMUNITY" },
+      { id: "impact-tag-id", text: "IMPACT" },
+    ],
+  },
+];
```
Same observation as this comment -> https://github.com/kindfi-org/kindfi/pull/82#discussion_r1929893420

AndlerRL commented on apps/web/components/sections/home/hero.tsx:
```diff
@@ -0,0 +1,148 @@
+            animate={{ opacity: 1, y: 0 }}
+            transition={{ duration: 0.5, delay: 0.6 }}
+          >
+            <Button size="lg" className="gradient-btn text-white">
+              Support with Crypto
+            </Button>
+            <Button
+              size="lg"
+              variant="outline"
+              className="gradient-border-btn"
+            >
+              Explore Causes
+            </Button>
```
@coderabbitai next time try to recommend less redundant changes. ARIA label is better to be used whenever there is no enough context. The application of these labels can be very redundant for the SR user, since the CTA has already enough text to understand what it will do or does since the parent and siblings can provide such context. Make sure to think step-by-step before landing into a conclusion.

AndlerRL commented on apps/web/lib/constants/animations.ts:
```diff
@@ -0,0 +1,28 @@
+  tap: { scale: 0.95 },
+};
+
+export const fadeInUpAnimation = {
+  initial: { opacity: 0, y: 20 },
+  whileInView: { opacity: 1, y: 0 },
+  viewport: { once: true },
+  transition: { duration: 0.6 }
+};
```
I suggest these times between animations (eye friendly):

1. `0.14`
2. `0.24`
3. `0.42`
4. `0.52`
5. `0.73`
6. `1.14`

I think `0.52` would do!

AndlerRL commented on apps/web/lib/constants/animations.ts:
```diff
@@ -0,0 +1,28 @@
+export const fadeInUp = {
+  initial: { opacity: 0, y: 20 },
+  animate: { opacity: 1, y: 0 },
+  transition: { duration: 0.5 },
+};
+
+export const staggerChildren = {
+  animate: { transition: { staggerChildren: 0.1 } },
+};
+
+export const badgeVariants = {
+  initial: { opacity: 0, y: 20 },
+  animate: { opacity: 1, y: 0, transition: { duration: 0.5 } },
+  hover: { scale: 1.05, transition: { duration: 0.2 } },
+  tap: { scale: 0.95 },
+};
+
+export const fadeInUpAnimation = {
+  initial: { opacity: 0, y: 20 },
+  whileInView: { opacity: 1, y: 0 },
+  viewport: { once: true },
+  transition: { duration: 0.6 }
+};
+
+export const fadeInUpVariants = {
+  hidden: { opacity: 0, y: 20 },
+  visible: { opacity: 1, y: 0 }
+};
```
Good catch! We should consider the reduced motion for the motion sickness a11y.

derianrddev commented:
@AndlerRL Thanks for the feedback and guidance! 🙌 I’ve addressed the requested changes:

- Moved the TS definitions from the mock-data folder into `~/lib/types/home.types.ts` as suggested.
- Reviewed and followed @coderabbitai's suggestions, ensuring all required ones are resolved.

Let me know if there’s anything else you’d like me to adjust!

Bran18 reviewed (changes requested):
@derianrddev i notice theres error related with the build in your branch, can you fix it so we can merge your changes? 

- `bun run build` should show you where the errors are. 

![Screenshot 2025-01-28 at 9 46 10 PM](https://github.com/user-attachments/assets/86cd3bbf-079c-4d58-8adc-03292618c9e0)

CC. @AndlerRL

derianrddev commented:
@Bran18 Ready!
![image](https://github.com/user-attachments/assets/43ad2da6-dfe6-4998-8ece-734b7b7c2b6e)

AndlerRL reviewed (approved):
LGTM! Very good placement on the various sections and the lazy loading has been tested and there was an improvement on the first paint :+1: merging!
